powerscale/provider/synciq_replication_rule_datasource.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/constants"
+	"terraform-provider-powerscale/powerscale/helper"
+	"terraform-provider-powerscale/powerscale/models"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &SyncIQRuleDataSource{}
+
+// NewSyncIQRuleDataSource creates a new data source.
+func NewSyncIQRuleDataSource() datasource.DataSource {
+	return &SyncIQRuleDataSource{}
+}
+
+// SyncIQRuleDataSource defines the data source implementation.
+type SyncIQRuleDataSource struct {
+	client *client.Client
+}
+
+// Metadata describes the data source arguments.
+func (d *SyncIQRuleDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_synciq_rule"
```

Comment:
Do we need replication in the datasource name?

Reply:
The GUI shows "Performance Rules" in its tab.
But the corresponding ansible module was named "dellemc.powerscale.synciqrules". It may be good to double check.